### PR TITLE
Enforce monetization actions to be in article response

### PR DESCRIPTION
**Pull request by justinchou-google**
b/291598203

Previously, revenue model was required in the code snippet. After the Premonetization launch, the `autoPromptType` parameter was deprecated from the code snippet, and the revenue model was inferred from the article response. As such, swg.js still supported both cases: when `Contribution` or `Subscription` was a valid action in the article, and when it was not. 

This change simplifies `autoPromptManager` and removes the backwards compatibility; all Contribution and Subscription publishers are assumed to have respective valid Contribution and Subscription actions in their article responses. If the `autoPromptType` param is provided, this is used to denote the mini prompt api is eligible.

### CODE REVIEW

File: src/runtime/auto-prompt-manager.ts [hunk 1]
```diff
@@ -494,60 +512,26 @@ export class AutoPromptManager {
...
-      this.interventionDisplayed_ = firstAction;
-      return firstAction;
-    }
-
-    // Suppress previously dismissed prompts.
-    let previouslyShownPrompts: string[] = [];
-    if (dismissedPrompts) {
-      previouslyShownPrompts = dismissedPrompts.split(',');
+    if (!this.isSubscription_(autoPromptType) && !this.pageConfig_.isLocked()) {
+      // Filter out contribution prompt if page is not paywalled.
```
Comment (oyj9109) [inline src/runtime/auto-prompt-manager.ts:516]: The assumption here is contribution is frequency capped in the earlier with `shouldShowMonetizationPromptAsSoftPaywall` being `false` at this point, right?
Would you mind explaining a little more about it in the comment?

File: src/runtime/auto-prompt-manager.ts [hunk 2]
```diff
@@ -494,60 +512,26 @@ export class AutoPromptManager {
...
-    );
-    // If autoprompt should be shown, and the contribution action is either the first action or
-    // not passed through audience actions, honor it and display the contribution prompt.
-    if (shouldShowMonetizationPromptAsSoftPaywall && contributionIndex < 1) {
-      this.interventionDisplayed_ = {type: TYPE_CONTRIBUTION};
-      return undefined;
+      // Suppress previously dismissed prompts.
+      let previouslyShownPrompts: string[] = [];
+      if (dismissedPrompts) {
+        previouslyShownPrompts = dismissedPrompts.split(',');
```
Comment (oyj9109) [inline src/runtime/auto-prompt-manager.ts:524]: nit: `const previouslyShownPrompts` as it's used within the if statement scope.

**Review by oyj9109 — COMMENTED** [review]
Sorry for delay in review. Overall change looks good. I especially like the part `getAction_` started returnning monetization type actions.
Left a few comments for nit and I'll continue reviewing tests. Thanks!

File: src/runtime/auto-prompt-manager.ts [hunk 3]
```diff
@@ -468,6 +479,13 @@ export class AutoPromptManager {
     shouldShowMonetizationPromptAsSoftPaywall?: boolean;
   }): Promise<Intervention | void> {
     const audienceActions = article.audienceActions?.actions || [];
+    if (shouldShowMonetizationPromptAsSoftPaywall) {
+      const action = (this.interventionDisplayed_ = audienceActions.filter(
```
[the remark this answers is not in the document]
Comment (justinchou-google) [reply src/runtime/auto-prompt-manager.ts:483]: This assignment is redundant, removed the initial `this.interventionDisplayed_` assignment.

Note: with the new frequency capping rules, dismissals (and hence `this.interventionDisplayed_`) will be deprecated.

File: src/runtime/auto-prompt-manager.ts [hunk 4]
```diff
@@ -243,12 +246,12 @@ export class AutoPromptManager {
 
     if (
       shouldShowMonetizationPromptAsSoftPaywall &&
-      potentialAction === undefined
+      this.isMonetizationAction_(potentialAction?.type)
     ) {
       this.deps_.win().setTimeout(() => {
         this.monetizationPromptWasDisplayedAsSoftPaywall_ = true;
```
[the remark this answers is not in the document]
Comment (justinchou-google) [reply src/runtime/auto-prompt-manager.ts:252]: Previously, `this.getAction_` would return `undefined` when contribution or subscription prompt should be displayed. I updated this fn to always return an action if an action is to be displayed.

File: src/runtime/auto-prompt-manager.ts [hunk 5]
```diff
@@ -177,20 +177,25 @@ export class AutoPromptManager {
...
+
     const hasValidEntitlements = entitlements.enablesThis();
     if (hasValidEntitlements) {
       return;
     }
 
-    // Override autoPromptType if it is undefined.
-    params.autoPromptType ??= this.getAutoPromptType_(
-      article?.audienceActions?.actions
+    const autoPromptType = this.getAutoPromptType_(
```
[the remark this answers is not in the document]
Comment (justinchou-google) [reply src/runtime/auto-prompt-manager.ts:189]: How does this sound?

File: src/runtime/auto-prompt-manager.ts [hunk 6]
```diff
@@ -243,12 +246,12 @@ export class AutoPromptManager {
...
     if (
       shouldShowMonetizationPromptAsSoftPaywall &&
-      potentialAction === undefined
+      this.isMonetizationAction_(potentialAction?.type)
     ) {
       this.deps_.win().setTimeout(() => {
         this.monetizationPromptWasDisplayedAsSoftPaywall_ = true;
         this.showPrompt_(
-          this.getPromptTypeToDisplay_(params.autoPromptType),
+          this.getPromptTypeToDisplay_(autoPromptType),
```
[the remark this answers is not in the document]
Comment (justinchou-google) [reply src/runtime/auto-prompt-manager.ts:254]: Great suggestion! done.

File: src/runtime/auto-prompt-manager.ts (hunk 1 above)
Comment (justinchou-google) [reply src/runtime/auto-prompt-manager.ts:516]: Done.

File: src/runtime/auto-prompt-manager.ts (hunk 2 above)
Comment (justinchou-google) [reply src/runtime/auto-prompt-manager.ts]: Good catch.

**Review by oyj9109 — APPROVED** [review]
Sorry again for delay in review and thanks for the change! Looking forward to upcoming refactoring with our frequency work.

**Pull request merged**